lib/core/engine/iteration.js

```diff
@@ -160,6 +161,20 @@ class Iteration {
         await preScript(context, commands);
       }
 
+      /*
+       * Look through the scripts and try to do our pre scripts.
+       */
+      if (this.scriptsByCategory) {
+        await browser.runScripts(this.scriptsByCategory, false, ScriptType.PRE);
+      }
+      if (this.asyncScriptsByCategory) {
+        await browser.runScripts(
+          this.asyncScriptsByCategory,
+          true,
+          ScriptType.PRE
+        );
+      }
+
       if (options.preURL) {
```

Comment:
Is it clear that we should run the pre scripts before `preURL`?  Kinda feels like no, we should run them after.

Reply:
Since the pre URL might affect the state of the browser and potentially contaminate any statistics that are tracked after the execution of the pre script, I 100% agree.